File: pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

#define Q 25
#define D 10
#define N 5
#define P 1 


int main(void)
{
    float change;
    do 
    {
        change = get_float("Change owned: ");
    }
    while (change < 0);
    
    int cents = round(change * 100);
    int owned = 0;

    int numquarts = cents / Q;
    cents -= numquarts * Q;
    owned += numquarts;

    int numdimes = cents / D;
    cents -= numdimes * D;
    owned += numdimes;

    int numnickels = cents / N;
    cents -= numnickels * N;
    owned += numnickels;
    
    int numpennies = cents / P;
    cents -= numpennies * P;
    owned += numpennies;
    
    printf("%i\n", owned);
}

File: pset6/cash/cash.py
from cs50 import get_float

Q = 25
D = 10
N = 5
P = 1 

while True:
    change = get_float("Change owned: ")
    if change >= 0:
        break

cents = round(change * 100)
owned = 0

numquarts = int(cents / Q)
cents -= int(numquarts * Q)
owned += int(numquarts)

numdimes = int(cents / D)
cents -= int(numdimes * D)
owned += int(numdimes)

numnickels = int(cents / N)
cents -= int(numnickels * N)
owned += int(numnickels)
    
numpennies = int(cents / P)
cents -= int(numpennies * P)
owned += int(numpennies)
    
print(owned)